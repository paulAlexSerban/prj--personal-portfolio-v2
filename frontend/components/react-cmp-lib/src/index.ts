import PersonalPortfolioWebpage from './templates/PersonalPortfolioWebpage';
export { PersonalPortfolioWebpage };
