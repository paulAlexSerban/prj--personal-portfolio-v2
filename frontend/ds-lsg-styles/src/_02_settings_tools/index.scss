@forward './_01_variables/index.scss';
@forward './functions/index.scss';
@forward './mixins/index.scss';
@forward './foundations/index.scss';
