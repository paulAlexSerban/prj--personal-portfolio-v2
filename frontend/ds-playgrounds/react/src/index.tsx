import React from 'react';

import { createRoot } from 'react-dom/client';
import { Paragraph } from '@prj--personal-portfolio-v2/react-cmp-lib/lib/index';

import '@prj--personal-portfolio-v2/ds-lsg-styles/lib/system/pages/main.css';
// import paragraphStyles from '@prj--personal-portfolio-v2/ds-lsg-styles/lib/components/atoms/paragraph.module.css';

const container = document.getElementById('root');
const root = createRoot(container!); // createRoot(container!) if you use TypeScript

root.render(
    <>
        <h2>Typography</h2>
        <h3>Paragraph Sizes</h3>
        <Paragraph styles={paragraphStyles} size="xs">
            This paragraph is XS - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>
        <Paragraph styles={paragraphStyles} size="sm">
            This Paragraph is SM - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>
        <Paragraph styles={paragraphStyles} size="base">
            This Paragraph is Base - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>
        <Paragraph styles={paragraphStyles} size="lg">
            This Paragraph is LG - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>
        <Paragraph styles={paragraphStyles} size="xl">
            This Paragraph is XL - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>

        <h3>Typography Styles</h3>

        <Paragraph styles={paragraphStyles} size="base" weight="light">
            This Paragraph is light - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>

        <Paragraph styles={paragraphStyles} size="base" weight="normal">
            This Paragraph is regular - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>

        <Paragraph styles={paragraphStyles} size="base" weight="bold">
            This Paragraph is bold - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>

        <Paragraph styles={paragraphStyles} size="base" weight="black">
            This Paragraph is black - Lorem ipsum dolor sit amet consectetur adipisicing elit. At, accusantium?
        </Paragraph>
    </>
);
