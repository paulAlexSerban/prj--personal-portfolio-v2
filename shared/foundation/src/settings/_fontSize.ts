/**
 * Font size
 */
const fontSizes: {
    xs: 'xs';
    sm: 'sm';
    base: 'base';
    lg: 'lg';
    xl: 'xl';
} = {
    xs: 'xs',
    sm: 'sm',
    base: 'base',
    lg: 'lg',
    xl: 'xl',
};
/**
 * Object.freeze() prevents the object from being modified making it immutable.
 */
export default Object.freeze(fontSizes);
