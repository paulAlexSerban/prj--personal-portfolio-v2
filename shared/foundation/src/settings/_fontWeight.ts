const fontWeight: {
    thin: 'thin';
    light: 'light';
    normal: 'normal';
    bold: 'bold';
    black: 'black';
} = {
    thin: 'thin',
    light: 'light',
    normal: 'normal',
    bold: 'bold',
    black: 'black',
};
/**
 * Object.freeze() prevents the object from being modified making it immutable.
 */
export default Object.freeze(fontWeight);
