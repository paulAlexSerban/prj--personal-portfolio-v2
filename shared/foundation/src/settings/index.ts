import fontSize from './_fontSize';
import fontWeight from './_fontWeight';
import spacing from './_spacing';

export { fontSize, fontWeight, spacing };
