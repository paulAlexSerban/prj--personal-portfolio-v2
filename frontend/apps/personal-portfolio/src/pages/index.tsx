import Head from 'next/head';
import Image from 'next/image';
import { Inter } from 'next/font/google';

import { Paragraph } from '@prj--personal-portfolio-v2/react-cmp-lib/lib';
import paraCss from '@prj--personal-portfolio-v2/ds-lsg-styles/lib/components/atoms/paragraph.module.css';

const inter = Inter({ subsets: ['latin'] });

export default function Home() {
    return (
        <>
            <Head>
                <title>Personal Portfolio v2</title>
                <meta name="description" content="Generated by create next app" />
                <meta name="viewport" content="width=device-width, initial-scale=1" />
                <link rel="icon" href="/favicon.ico" />
            </Head>
            <main>
                <Paragraph className={[paraCss.paragraph, paraCss['paragraph--base']].join(' ')}>
                    BASE - Lorem ipsum, dolor sit amet consectetur adipisicing elit. Incidunt, nemo.
                </Paragraph>
            </main>
        </>
    );
}
