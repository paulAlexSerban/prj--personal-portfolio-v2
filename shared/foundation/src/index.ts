import { randomChoice } from './arrays';

import { padToThree } from './numbers';

import { fontSize, spacing, fontWeight } from './settings';

import { randomWord } from './tools';

export { randomChoice, padToThree, fontSize, spacing, randomWord, fontWeight };
