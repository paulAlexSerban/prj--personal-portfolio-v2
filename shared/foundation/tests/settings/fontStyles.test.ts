import { fontWeight } from '../../src/settings';

test('snapshot of fontWeight', () => {
    expect(fontWeight).toMatchSnapshot();
});
