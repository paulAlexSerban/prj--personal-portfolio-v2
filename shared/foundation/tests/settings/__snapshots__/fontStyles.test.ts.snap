// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of fontWeight 1`] = `
{
  "black": "black",
  "bold": "bold",
  "light": "light",
  "normal": "normal",
  "thin": "thin",
}
`;
